src/write/macho.rs

```diff
@@ -590,6 +590,28 @@ impl<'a> Object<'a> {
                             (RelocationKind::Absolute, RelocationEncoding::Generic, 0) => {
                                 (false, macho::ARM64_RELOC_UNSIGNED)
                             }
+                            // Due to the fixup an addend of -4 here means the original addend was 0
+                            (RelocationKind::Relative, RelocationEncoding::AArch64Call, -4) => {
+                                (true, macho::ARM64_RELOC_BRANCH26)
+                            }
+                            // Non-zero addend, so we have to encode the addend separately
+                            (RelocationKind::Relative, RelocationEncoding::AArch64Call, value) => {
+                                // first emit the BR26 relocation
+                                let reloc_info = macho::RelocationInfo {
+                                    r_address: reloc.offset as u32,
+                                    r_symbolnum,
+                                    r_pcrel: true,
+                                    r_length,
+                                    r_extern: true,
+                                    r_type: macho::ARM64_RELOC_BRANCH26,
+                                };
+                                buffer.write(&reloc_info.relocation(endian));
+
+                                // set up a separate relocation for the addend
+                                // we add 4 to undo the fixup of -4
+                                r_symbolnum = (value + 4) as u32;
```

Comment:
Instead of adding 4 here, change `macho_fixup_relocation` so that this isn't needed.